core-api/src/main/java/com/optimizely/ab/decision/experiment/service/ExperimentBucketerDecisionService.java

```diff
@@ -0,0 +1,77 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+package com.optimizely.ab.decision.experiment.service;
+
+import com.optimizely.ab.bucketing.Bucketer;
+import com.optimizely.ab.config.Experiment;
+import com.optimizely.ab.config.Variation;
+import com.optimizely.ab.decision.entities.DecisionStatus;
+import com.optimizely.ab.decision.entities.ExperimentDecision;
+import com.optimizely.ab.decision.entities.Reason;
+import com.optimizely.ab.decision.experiment.IExperimentDecisionService;
+import com.optimizely.ab.event.internal.UserContext;
+
+import com.optimizely.ab.internal.ControlAttribute;
+import com.optimizely.ab.internal.ExperimentUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ * ExperimentBucketerDecisionService makes a decision using the experiment bucketer
+ */
+public class ExperimentBucketerDecisionService implements IExperimentDecisionService {
+
+    private final Bucketer bucketer = new Bucketer();
+    private static final Logger logger = LoggerFactory.getLogger(ExperimentBucketerDecisionService.class);
+
+    @Override
+    public ExperimentDecision getDecision(@Nonnull Experiment experiment,
+                                          @Nonnull UserContext userContext) {
+        String bucketingId = getBucketingId(userContext.getUserId(), userContext.getAttributes());
+        Variation variation = bucketer.bucket(experiment, bucketingId, userContext.getProjectConfig());
```

Comment:
I think need to revise this code. Since bucketer suppose to return only bucketed value. rest of the logic won't go there.